flow/layers/cacheable_layer.h

```diff
@@ -0,0 +1,106 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLOW_LAYERS_CACHEABLE_LAYER_H_
+#define FLUTTER_FLOW_LAYERS_CACHEABLE_LAYER_H_
+
+#include <memory>
+#include "flutter/flow/embedded_views.h"
+#include "flutter/flow/layers/container_layer.h"
+#include "flutter/flow/raster_cache.h"
+#include "flutter/flow/raster_cacheable_entry.h"
+#include "include/core/SkColor.h"
+#include "include/core/SkMatrix.h"
+#include "include/core/SkPoint.h"
+#include "include/core/SkRect.h"
+
+namespace flutter {
+
+class Cacheable {
```

Comment:
The way this class is mixed in here, it just becomes a dispatcher for 3 other classes that encapsulate the concepts for each of the 3 types of cacheable.

Just instantiate the type of CacheableItem in the constructor of the layers and declare the field in the in those classes and this entire class goes away.

AutoCache can be a simple class that takes that object and calls methods on it and doesn't need 3 different type of Create methods.